ocaml/xapi/quicktest_raw_vdi_import.ml

```diff
@@ -0,0 +1,60 @@
+(*
+ * Copyright (C) 2013 Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+(* Check the VDI.copy support for copying disk deltas *)
+
+open Sparse_encoding
+open Client
+open Quicktest_common
+open Http
+
+let http request f =
+  let open Xmlrpc_client in
+  let transport =
+    if !using_unix_domain_socket
+    then Unix Xapi_globs.unix_domain_socket
+    else SSL(SSL.make ~use_fork_exec_helper:false (), !host, 443) in
+  with_transport transport (with_http request f)
+
+let import_raw_vdi ~session_id ~task_id f =
+  (*let uri = Printf.sprintf "/import_raw_vdi?session_id=%s&vdi=%s&task_id=%si&chunked=true"
+    (Ref.string_of session_id) (Ref.string_of vdi) (Ref.string_of task_id) in *)
+  let req = Http.Request.make ~version:"1.0"
+      ~user_agent:"quicktest"
+      ~query:[
+        "session_id", Ref.string_of session_id;
+        "sr_id","fake_sr";
+        "task_id", Ref.string_of task_id;
+        "chunked", "true"
+      ]
+      Http.Put "/import_raw_vdi" in
+  http req (fun (_, fd) -> f fd)
+
+let start session_id =
+  let t = make_test "import_raw_vdi_quicktest" 1 in
+  start t;
+  let task_id = Client.Task.create ~rpc:!rpc ~session_id
+      ~label:"quicktest import raw vdi"
+      ~description:"" in
+  try   
+    import_raw_vdi ~session_id ~task_id (fun fd ->())
+  with e -> 
+    let a = Client.Task.get_record ~rpc:!rpc ~session_id ~self:task_id in
+    Client.Task.destroy ~rpc:!rpc ~session_id ~self:task_id;
+    if a.API.task_status <> `failure 
```

Comment:
If there are multiple cases, it might be better to match them and to provide a more detailed error message for each.